src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,133 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * The Parser class represents a parser that takes inputs and produces an output.
+ */
+public class Parser {
+
+    /**
+     * Processes and responds to the inputs received.
+     *
+     * @param input Input typed by user.
+     * @param tasks List of tasks.
+     * @param storage Stores the file and handles file methods.
+     */
+    public static void parseInput(String input, TaskList tasks, Storage storage) {
```

Comment:
perhaps you may want to call .trim() on input first before the if else loops - to account for edge cases. For example if I key in "   list    " instead of "list".